pubtools/_pulp/tasks/common.py

```diff
@@ -182,3 +190,161 @@ def publish_with_cache_flush(
         out.append(flush_ud)
 
         return out
+
+
+class PulpRepositoryOperation(CDNCache, UdCache, PulpTask):
+    def __init__(self):
+        super(PulpRepositoryOperation, self).__init__()
+
+        self.task_state = "PENDING"
+
+    def add_args(self):
+        super(PulpRepositoryOperation, self).add_args()
+
+        self.parser.add_argument(
+            "--skip", help="skip given comma-separated sub-steps", type=str
+        )
+
+    def fail(self, *args, **kwargs):
+        LOG.error(*args, **kwargs)
+        sys.exit(30)
+
+    def log_remove(self, removed_repo):
+        return self.log_repo_action(removed_repo, "removed")
+
+    def log_copy(self, copied_repo):
+        return self.log_repo_action(copied_repo, "copied")
+
+    def log_repo_action(self, repo, action):
+        # Given a repo which has been altered, log some messages
+        # summarizing the affected unit(s).
+        content_types = {}
+
+        for task in repo.tasks:
+            for unit in task.units:
+                type_id = unit.content_type_id
+                content_types[type_id] = content_types.get(type_id, 0) + 1
+
+        task_ids = ", ".join(sorted([t.id for t in repo.tasks]))
+        repo_id = repo.repo.id
+        if not content_types:
+            LOG.warning("%s: no content removed, tasks: %s", repo_id, task_ids)
+        else:
+            removed_types = []
```

Comment:
Similar to the above comment, since we don't know if this was actually a remove operation, some more general term like "affected_types" may be more appropriate and less confusing to the reader.